subsys/bluetooth/common/Kconfig

```diff
@@ -214,13 +214,19 @@ config BT_HCI_VS_EXT
 	  Host and/or Controller. This enables  Write BD_ADDR, Read Build Info,
 	  Read Static Addresses and Read Key Hierarchy Roots vendor commands.
 
+config BT_HCI_VS_EVT
```

Comment:
would gating some other kconfig options behind that one make sense? right now it seems it's only used for code.

Reply:
or at least add `select`s or `depends on` for the events that depend on that.

Reply:
This option is added for all possible events that are and may be implemented in future. It is used to guard `vs_events_mask` and `vs_event`. That prevents build time warnings if vendor specific events are not used. Also allows to select only required events to be enabled. 

It is already selected by `BT_CTLR_DF_VS_CL_IQ_REPORT_16_BITS_IQ_SAMPLES`, `BT_CTLR_DF_VS_CONN_IQ_REPORT_16_BITS_IQ_SAMPLES`, `BT_DF_VS_CL_IQ_REPORT_16_BITS_IQ_SAMPLES` and BT_DF_VS_CONN_IQ_REPORT_16_BITS_IQ_SAMPLES`.

Reply:
yeah but you have more VS events that that, see the mask you made.

Reply:
yes there are other events defined in `hci_vs.h` where:
- `bt_hci_evt_vs_fatal_error` is a bit in parallel to other because it uses its private memory for `net_buf`. Needed to be sure that there is a free buf to send event e.g. in case of Hard Fault.
- `bt_hci_evt_vs_trace_info`, `bt_hci_evt_vs_scan_req_rx` are not used by Controller or Host. They are just defined... At least I didn't find any code that uses it.